metrics-server/src/lib.rs

```diff
@@ -59,20 +59,44 @@ pub mod server;
 pub mod metrics;
 pub mod error;
 
-pub fn metrics_server<P, CM>(consensus: Arc<Consensus<P>>, ip: IpAddr, port: u16, password: Option<String>) -> Result<Box<dyn Future<Item=(), Error=()> + Send + Sync>, Error>
-    where P: ConsensusProtocol + 'static,
-          CM: AbstractChainMetrics<P> + server::Metrics + 'static
-{
-    Ok(Box::new(Server::try_bind(&SocketAddr::new(ip, port))?
-        .serve(move || {
-            server::MetricsServer::new(
-                vec![
-                    Arc::new(CM::new(consensus.blockchain.clone())),
-                    Arc::new(MempoolMetrics::new(consensus.mempool.clone())),
-                    Arc::new(NetworkMetrics::new(consensus.network.clone()))
-                ],
-                attributes!{ "peer" => consensus.network.network_config.peer_address() },
-            password.clone())
+
```

Comment:
Not related to your code, but as a reminder: Jeff should port his changes from core-rs that switch the metrics server to HTTPS.